src/core/public/notifications/toasts/toasts_api.tsx

```diff
@@ -33,13 +35,20 @@ import { OverlayStart } from '../../overlays';
  *
  * @public
  */
-export type ToastInputFields = Pick<Toast, Exclude<keyof Toast, 'id'>>;
+export type ToastInputFields = Pick<EuiToast, Exclude<keyof EuiToast, 'id' | 'text' | 'title'>> & {
+  title?: string | MountPoint;
+  text?: string | MountPoint;
+};
+
+export type Toast = ToastInputFields & {
+  id: string;
+};
 
 /**
  * Inputs for {@link IToasts} APIs.
  * @public
  */
-export type ToastInput = string | ToastInputFields | Promise<ToastInputFields>;
+export type ToastInput = string | ToastInputFields;
```

Comment:
The `promise<>` was unused, actually broken (somewhere in the code we object spread `ToastInput` assuming that's not a promise), and made some conversion quite hard, to I removed them. (and I don't think that really makes sense, in case of async toast, the waiting can be done in the caller code)